internal/testing/fakeopenai/requests.go

```diff
@@ -0,0 +1,240 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package fakeopenai
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+)
+
+// Cassette names for testing..
+const (
+	CassetteChatBadModel      = "chat-bad-model"
+	CassetteChatBasic         = "chat-basic"
+	CassetteChatJSONMode      = "chat-json-mode"
+	CassetteChatMultimodal    = "chat-multimodal"
+	CassetteChatMultiturn     = "chat-multiturn"
+	CassetteChatNoMessages    = "chat-no-messages"
+	CassetteChatParallelTools = "chat-parallel-tools"
+	CassetteChatStreaming     = "chat-streaming"
+	CassetteChatTools         = "chat-tools"
+	CassetteChatUnknownModel  = "chat-unknown-model"
+	CassetteEdgeBadRequest    = "edge-bad-request"
+	CassetteEdgeBase64Image   = "edge-base64-image"
+)
+
+// NewRequest creates a new HTTP request for the given cassette name.
+// It automatically sets the method to POST, content type to application/json,
+// and includes the X-Cassette-Name header for cassette selection.
+func NewRequest(baseURL, cassetteName string) (*http.Request, error) {
+	// Get the request body for this cassette..
+	requestBody, ok := requestBodies[cassetteName]
+	if !ok {
+		return nil, fmt.Errorf("unknown cassette name: %s", cassetteName)
+	}
+
+	// Create the request..
+	req, err := http.NewRequest(http.MethodPost, baseURL+"/chat/completions", bytes.NewReader([]byte(requestBody)))
+	if err != nil {
+		return nil, err
+	}
+
+	// Set headers..
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Cassette-Name", cassetteName)
+
+	return req, nil
+}
+
+// requestBodies contains the actual request body for each cassette..
+// These are kept private as they should only be accessed via NewRequest..
+var requestBodies = map[string]string{
```

Comment:
this is a relic. I'll update to grab it from the cassette directly.

Reply:
not a relic. I will update the comment. This is needed for re-recording ;) (you shouldn't depend on the cassette itself to know what request you should record!